File: 3917.count-indices-with-opposite-parity/solution.kt
class Solution {
    fun countOppositeParity(nums: IntArray): IntArray {
        val res = IntArray(nums.size)
        var even = 0
        var odd = 0

        for (i in nums.lastIndex downTo 0) {
            if (nums[i] % 2 == 0) {
                res[i] = odd
                even++
            } else {
                res[i] = even
                odd++
            }
        }

        return res
    }
}

File: 3917.count-indices-with-opposite-parity/solution.go
func countOppositeParity(nums []int) []int {
    odd, even := 0, 0
    answer := make([]int, len(nums))

    for i := len(nums)-1; i >= 0; i-- {
        if nums[i]%2 == 0 {
            answer[i] = odd
            even++
        } else {
            answer[i] = even
            odd++
        }
    }

    return answer
}